test/shared/src/main/scala-2/zio/test/SmartAssertMacros.scala

```diff
@@ -173,10 +173,10 @@ class SmartAssertMacros(val c: blackbox.Context) {
       case q"!($inner)" =>
         AST.Not(parseExpr(inner), pos.getPos(tree), pos.getPos(inner))
 
-      case q"$lhs && $rhs" =>
+      case q"$lhs && $rhs" if lhs.tpe == typeOf[Boolean] =>
```

Comment:
This makes sure that if we have a `&&` function defined for some type we gonna still handle that in assert true:
```
object AssertSpec extends ZIOSpecDefault {
  override def spec: Spec[Any, Any] =
    test("") {

      val first = TestClazz(10)
      val second = TestClazz(20)
      zio.test.assertTrue(first && second)
    }
}
case class TestClazz(int: Int){
  def &&(anotherClz: TestClazz): Boolean = {
    false
  }
}
```
Currently, we are just going to fail compilation for that case.

Reply:
Again, open to propositions if there is a simpler/prettier way to write this.